chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -243,28 +253,48 @@ impl NetworkState {
                 return Err(RegisterPeerError::Banned);
             }
 
-            if conn.peer_type == PeerType::Inbound {
-                if !this.is_inbound_allowed(&peer_info) {
-                    // TODO(1896): Gracefully drop inbound connection for other peer.
-                    let tier2 = this.tier2.load();
-                    tracing::debug!(target: "network",
-                        tier2 = tier2.ready.len(), outgoing_peers = tier2.outbound_handshakes.len(),
-                        max_num_peers = this.max_num_peers.load(Ordering::Relaxed),
-                        "Dropping handshake (network at max capacity)."
-                    );
-                    return Err(RegisterPeerError::ConnectionLimitExceeded);
+            match conn.tier {
+                tcp::Tier::T1 => {
+                    if !this.config.tier1.as_ref().map_or(false, |c| c.enable_inbound) {
```

Comment:
shouldn't this be under the if below (if peer_type == Inbound?)

Reply:
true, thank you